cfn-resources/teams/cmd/resource/team-user/team_user.go

```diff
@@ -0,0 +1,106 @@
+// Copyright 2023 MongoDB Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package teamuser
+
+import (
+	"context"
+	"net/http"
+
+	"github.com/mongodb/mongodbatlas-cloudformation-resources/util"
+	"github.com/mongodb/mongodbatlas-cloudformation-resources/util/logger"
+	atlasv2 "go.mongodb.org/atlas-sdk/v20231001001/admin"
+)
+
+func FilterOnlyValidUsernames(mongoDBCloudUsersAPIClient TeamUsersAPI, usernames []string) ([]atlasv2.CloudAppUser, *http.Response, error) {
+	var validUsers []atlasv2.CloudAppUser
+	for _, elem := range usernames {
+		userToAdd, httpResp, err := mongoDBCloudUsersAPIClient.GetUserByUsername(context.Background(), elem)
+		if err != nil {
+			_, _ = logger.Warnf("Error while getting the user %s: (%+v) \n", elem, err)
+			return nil, httpResp, err
+		}
+		validUsers = append(validUsers, *userToAdd)
+	}
+	return validUsers, nil, nil
+}
+
+func initUserSet(users []atlasv2.CloudAppUser) map[string]bool {
+	usersSet := make(map[string]bool)
+	for i := 0; i < len(users); i++ {
+		usersSet[*(users[i]).Id] = true
+	}
+	return usersSet
+}
+
+func GetUsersToAddAndRemove(currentUsers []atlasv2.CloudAppUser, newUsers []atlasv2.CloudAppUser) (toAdd []string, toDelete []string, err error) {
+	// Create two sets to store the elements in A and B
+	currentUsersSet := initUserSet(currentUsers)
+	newUsersSet := initUserSet(newUsers)
+
+	// Create two arrays to store the elements to be added and deleted
+	toAdd = []string{}
+	toDelete = []string{}
+
+	// Iterate over the elements in B and add them to the toAdd array if they are not in A
+	for elem := range newUsersSet {
+		if _, ok := currentUsersSet[elem]; !ok {
+			toAdd = append(toAdd, elem)
+		}
+	}
+
+	// Iterate over the elements in A and add them to the toDelete array if they are not in B
+	for elem := range currentUsersSet {
+		if _, ok := newUsersSet[elem]; !ok {
+			toDelete = append(toDelete, elem)
+		}
+	}
+
+	// Return the two arrays
+	return toAdd, toDelete, nil
+}
+
+func UpdateTeamUsers(teamUsersAPIService TeamUsersAPI, existingTeamUsers *atlasv2.PaginatedApiAppUser, usernames []string, orgID, teamID string) error {
+	var newUsers []atlasv2.AddUserToTeam
+
+	validUsernames, _, err := FilterOnlyValidUsernames(teamUsersAPIService, usernames)
+	if err != nil {
+		return err
+	}
+	usersToAdd, usersToRemove, err := GetUsersToAddAndRemove(existingTeamUsers.Results, validUsernames)
```

Comment:
listing is paginated and I see you are doing the call with the default of 100 per page, have you considered the case of a team having more than 100 users?

Reply:
That's a good call. While I would love to further improve this, I think this is another bug/behaviour from the original code. I am creating a follow-up task to update that accordingly.

https://jira.mongodb.org/browse/INTMDB-1280

Resolving, feel free to re-open if you have concerns.